### PR TITLE
Feat: Create delete requests for notification service

### LINKED ISSUE #62: Fix: handle optional fields correctly in proto
Protobuf sets a default value for null such as:
* "" for string
* 0 for int
* 1970-01-01 for datetime
Handle it

### PR DESCRIPTION
**Pull request by ZixMai**
1.  Updated notification service proto file
2.  Added 4 delete request (by event and author ids, by events and author ids, by author id, by event id)
3.  Changed mock repository delete method
4.  Corrected docstrings in event and notification services

### CODE REVIEW

File: notification_service/repository/mock_notification_repository.py [hunk 1]
```diff
@@ -280,3 +290,121 @@ async def delete_notification_by_id(self, notification_id: str) -> None:
...
+            Can't delete notification with provided data.
+
+        """
+        indexes = tuple(
+            i
+            for i in range(len(self._notifications))
+            if self._notifications[i].event_id == event_id
+            and self._notifications[i].deleted_at is None
+        )
+        if indexes is None or len(indexes) == 0:
```
Comment (Nhsdkk) [inline notification_service/repository/mock_notification_repository.py]: Redundant check for None

File: notification_service/repository/mock_notification_repository.py [hunk 2]
```diff
@@ -280,3 +290,121 @@ async def delete_notification_by_id(self, notification_id: str) -> None:
...
+            Can't delete notification with provided data.
+
+        """
+        indexes = tuple(
+            i
+            for i in range(len(self._notifications))
+            if self._notifications[i].author_id == author_id
+            and self._notifications[i].deleted_at is None
+        )
+        if indexes is None or len(indexes) == 0:
```
Comment (Nhsdkk) [inline notification_service/repository/mock_notification_repository.py]: Redundant check for None

File: notification_service/repository/mock_notification_repository.py [hunk 3]
```diff
@@ -280,3 +290,121 @@ async def delete_notification_by_id(self, notification_id: str) -> None:
...
+
+        """
+        indexes = tuple(
+            i
+            for i in range(len(self._notifications))
+            if self._notifications[i].author_id == author_id
+            and self._notifications[i].event_id in event_ids
+            and self._notifications[i].deleted_at is None
+        )
+        if indexes is None or len(indexes) == 0:
```
Comment (Nhsdkk) [inline notification_service/repository/mock_notification_repository.py]: Redundant check for None

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Make changes

**Review by MaxPlays35 — APPROVED** [review]
Good

**Review by Nhsdkk — APPROVED** [review]
LGTM

**Pull request merged**